src/hooks/useShortcut.js

```diff
@@ -0,0 +1,99 @@
+import { useEffect, useCallback, useReducer } from 'react'
+
+const keysReducer = (state, action) => {
+  switch (action.type) {
+    case 'set-key-down':
+      const keydownState = { ...state, [action.key]: true }
+      return keydownState
+    case 'set-key-up':
+      const keyUpState = { ...state, [action.key]: false }
+      return keyUpState
+    case 'reset-keys':
+      const resetState = { ...action.data }
```

Comment:
Please consider using the initialState for a reset so that you don't have to provide it when dispatching an action.

Reply:
I'm not sure I fully understood the annotation, but action.data does contain the initialState in line 12